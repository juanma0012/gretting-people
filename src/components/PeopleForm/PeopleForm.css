.PeopleForm >  textarea {
    width: 100%;
    height: 100px;
    font-size: 1em;
    color: #333;
    box-sizing: border-box;
    letter-spacing: 1px;
    padding: 7px 15px;
    border: 1px solid #ccc;
    border-radius: 5px;
}