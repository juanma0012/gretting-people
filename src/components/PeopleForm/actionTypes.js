export const SET_PEOPLE = 'peopleForm/SET_PEOPLE';
