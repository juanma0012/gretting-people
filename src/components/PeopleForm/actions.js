import * as actionTypes from './actionTypes';
import * as greetingActions from '../GreetingList/actions';

export const setPeople = (peopleString) => {
    return (dispatch, getState) => {
        console.log("peopleString",peopleString);
        let people =  peopleString.split('\n').filter(item => item.trim() !== '' );
        dispatch({ type: actionTypes.SET_PEOPLE, people });
        let state = getState();
        let language = state.languageDropdown.language;
        dispatch(greetingActions.setGreetings(people, language));
    }
}
