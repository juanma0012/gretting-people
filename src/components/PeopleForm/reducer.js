import { resolveEach } from 'redux-resolver'
import { updateObject } from '../../shared/utility';
import * as actionTypes from './actionTypes'

const setPeople = (state, action) => {
    console.log("action", action)
    return updateObject(state, {
        people: action.people
    });
};

export default resolveEach([], {
    [actionTypes.SET_PEOPLE]: setPeople
})