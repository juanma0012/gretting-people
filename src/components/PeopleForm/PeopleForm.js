import { connect } from 'react-redux';
import React from 'react';
import PropTypes from 'prop-types';
import * as actions from './actions';

const PeopleForm = ({ onSetPeople }) => {
    let value;
    const onChange = e => {
        onSetPeople(e.target.value);
    }
    return (
        <div>
            <form>
                <textarea value={value} rows="3" cols="20" onChange={onChange}></textarea>
            </form>
        </div>
    )
};

PeopleForm.propTypes = {
    onSetPeople: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
    onSetPeople: (people) => dispatch(actions.setPeople(people))
});

export default connect(
    null,
    mapDispatchToProps
)(PeopleForm);
