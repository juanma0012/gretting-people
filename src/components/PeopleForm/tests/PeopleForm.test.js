import React from 'react';
import { shallow } from 'enzyme';
import { PeopleForm } from '../PeopleForm';

describe('Component PeopleForm ', () => {
    describe('WHEN is initialized ', () => {
        it('SHOULD render correctly ', () => {
            const component = shallow(<PeopleForm onSetPeople={() => { }} />);
            expect(component).toMatchSnapshot();
        });
    });

    describe('WHEN a name is written in the textarea input ', () => {
        const changeFn = jest.fn();
        it('SHOULD calls the onSetPeople method', () => {
            const component = shallow(<PeopleForm onSetPeople={changeFn} />);
            component.find('textarea').simulate('change', { target: { value: "Fake Event" } });
            expect(changeFn).toHaveBeenCalled();
        });
    });
});
