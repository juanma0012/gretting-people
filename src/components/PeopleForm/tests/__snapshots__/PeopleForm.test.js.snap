// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PeopleForm  WHEN is initialized  SHOULD render correctly  1`] = `
<div
  className="peopleForm"
>
  <textarea
    onChange={[Function]}
  />
</div>
`;
